idaes/core/solvers/petsc.py

```diff
@@ -517,10 +518,28 @@ def petsc_dae_by_time_element(
         between = pyo.Set(initialize=sorted(between))
         between.construct()
 
+    # Make sure that time is a ContinuousSet that has been discretized
+    try:
+        time.get_discretization_info()["nfe"]
+    except AttributeError:
+        raise RuntimeError("Argument time is not a Pyomo ContinuousSet")
+    except KeyError:
+        raise RuntimeError("The ContinuousSet time has not been discretized")
+
```

Comment:
Is this the best way to check for a discretized continuous set? I would probably have used `isinstance(time, ContinuousSet)` and `"nfe" in time.get_discretization_info()` to avoid catching arbitrary `AttributeError`/`KeyError`.

Reply:
I wanted to avoid using `isinstance` because it's a bit looked-down upon in Python programming---typically you want to just try doing what you want to do and ask for forgiveness later. In this case, however, it's probably better to just use it.

It would be better if `ContinuousSet` had a function `has_been_discretized` or similar to check directly rather than checking dictionary members.